featuretools/primitives/standard/binary_transform.py

```diff
@@ -495,7 +495,11 @@ class MultiplyNumeric(TransformPrimitive):
         [2, 2, 4]
     """
     name = "multiply_numeric"
-    input_types = [Numeric, Numeric]
+    input_types = [
+        [Numeric, Numeric],
+        [Numeric, Boolean],
+        [Boolean, Numeric]
+    ]
```

Comment:
can we also support Boolean * Boolean

Reply:
We can. If we add `[Boolean, Boolean]` to the input types it works. However, during testing, the following Warning is generated:
```
UserWarning: evaluating in Python space because the '*' operator is not supported by numexpr for the bool dtype, use '&' instead
```
I haven't traced this back fully to see where exactly it is coming from, or what it would take to eliminate the warning.  It seems to work fine though.

Reply:
Let me know how you would like to proceed here and I can update accordingly.

Reply:
We could add a MultiplyBoolean primitive to handle this, it would require updating how FeatureBase overrides the `__mul__` function

Reply:
@rwedge Pushing a new update right now with a `MultiplyBoolean` primitive added. Let me know if this is what you were thinking.

Reply:
@thehomebrewnerd Yes, that was what I was thinking